crates/host/src/wasmbus/providers/mod.rs

```diff
@@ -1,2 +1,359 @@
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Context as _;
+use async_nats::Client;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use bytes::Bytes;
+use cloudevents::EventBuilderV10;
+use futures::Future;
+use nkeys::XKey;
+use tokio::io::AsyncWriteExt;
+use tokio::sync::{broadcast, RwLock};
+use tokio::task::JoinSet;
+use tokio::{process, select};
+use tracing::{error, instrument, trace, warn};
+use wascap::jwt;
+use wasmcloud_core::{provider_config_update_subject, HealthCheckResponse};
+use wasmcloud_tracing::context::TraceContextInjector;
+
+use crate::wasmbus::{config::ConfigBundle, Annotations};
+use crate::wasmbus::{event, injector_to_headers};
+
 mod http_server;
 mod messaging_nats;
+
+/// An Provider instance
+#[derive(Debug)]
+pub(crate) struct Provider {
+    pub(crate) image_ref: String,
+    pub(crate) claims_token: Option<jwt::Token<jwt::CapabilityProvider>>,
+    pub(crate) xkey: XKey,
+    pub(crate) annotations: Annotations,
+    #[allow(unused)]
+    /// Config bundle for the aggregated configuration being watched by the provider
+    pub(crate) config: Arc<RwLock<ConfigBundle>>,
+    /// Whether the provider should be restarted on exit
+    pub(crate) restart_on_exit: Arc<AtomicBool>,
```

Comment:
Nit: I'd add some extra docs here. The naming kinda threw me off as it seems like this is a configurable value, but it is really only used for builtins as a configuration value. Otherwise I would have said this would make more sense as a variable called `is_shutting_down` or something like that

Reply:
Yep you got it, naming this one is a little weird but I will try to make it most descriptive. Essentially all I want to do is pause the provider restarts when it's being stopped

Reply:
Renamed to `shutdown`, which is an indicator of, well, the provider shutting down.